docs/rules/saga-direct-delegation.md

```diff
@@ -0,0 +1,58 @@
+# Enforce using `yield*` to delegate to other sagas
+# saga-direct-delegation
+
+This rule enforces that delegation to sagas is done directly using `yield*`, rather than indirectly by plain `yield` of the returned iterator to the saga middleware.
+
+This has the benefit of maintaining the delegator (caller) on the call stack, which aids debugging and gives better error stack traces.
+
+The rule is autofixable.
+
+## Examples of incorrect code for this rule:
+
+❌ Delegations not using `yield*`, where the callee is in the same file and accessed by identifier
+
+```js
+/* eslint saxo/saga-direct-delegation: "error" */
+function* delegate() {}
+
+function* caller() {
+    yield delegate();
+}
+```
+
+## Examples of correct code for this rule:
+
+✅ Delegations not using `yield*`, where the callee is imported from from another file
+
+```js
+/* eslint saxo/saga-direct-delegation: "error" */
+import delegate from './sagas';
+
+function* caller() {
+    yield delegate();
+}
+```
+
+✅ Delegations not using `yield*`, where the callee is not accessed by identifier
+
+```js
+/* eslint saxo/saga-direct-delegation: "error" */
+const obj = {
+    delegate: function* () {}
+};
+
+function* caller() {
+    yield obj.delegate();

```

Comment:
Agree, I came here now myself (not having seen your comment) to comment that I should make it clearer that this is a limitation. I had written it in the current format based on the `string-constant-types` doc, without thinking it through enough.

Reply:
64f4570 ... should address this.

I'm not too concerned about handling this particular `obj.delegate()` or similar cases in future, given typical saga patterns.

Delegating to an imported saga is however common. I think only a best-effort approach would be possible due to ESLint working on a per-file basis - we can't get the type of anything imported from another file. Something like guessing based on the module path from the import declaration... ending `/[Ss]agas\.js$/` or having a directory `/\/.*[Ss]agas\//` in its path. It wouldn't be universal or robust, and would be prone to false positives/negatives, so would need to have an option to enable/configure.

Reply:
Actually, on that last paragraph, thinking of some of the things eslint-plugin-import can do, it might be possible to do this properly rather than what I described. I'll have a look at how that plugin works sometime in the future.

Reply:
The import plugin does some complicated things with import maps/resolvers, I don't think it's reasonably feasible to implement that here. So back to the module path idea (previous comment), I don't think it's a bad one.